[🐛 Bug]: Usage Error: Couldn't find a script named "next-on-pages-vercel-cli".

### next-on-pages environment related information


System:
	Platform: darwin
	Arch: arm64
	Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
	CPU: (12) arm64 Apple M2 Max
	Memory: 32 GB
	Shell: /bin/zsh
Binaries:
	Node: 18.15.0
	Yarn: 3.6.0
	npm: 9.5.0
	pnpm: 8.6.1
Package Manager Used: yarn (berry)
Relevant Packages:
	@cloudflare/next-on-pages: 1.0.2
	vercel: [object Promise]
	next: [object Promise]



### Description

The next-on-pages doesn't work with yarn berry

~/gits/personal-site main !3 ❯ npx @cloudflare/next-on-pages                                                                                                                      20s

⚡️ @cloudflare/next-on-pages CLI v.1.0.2
⚡️ Detected Package Manager: yarn (berry)
⚡️ Preparing project...
⚡️ Project is ready
⚡️ Building project...
▲  Usage Error: Couldn't find a script named "next-on-pages-vercel-cli".
▲  
▲  $ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...

⚡️ The Vercel build (`yarn dlx vercel build`) command failed. For more details see the Vercel logs above.
⚡️ If you need help solving the issue, refer to the Vercel or Next.js documentation or their repositories.




### Reproduction

https://github.com/davisuga/personal-site

### Pages Deployment Method

Pages CI (GitHub/GitLab integration)

### Pages Deployment ID

_No response_

### Additional Information

_No response_

### Would you like to help?

- [ ] Would you like to help fixing this bug?

It fails with pnpm too



11:17:49.245 | Installing project dependencies: pnpm install
-- | --
11:17:51.036 | Lockfile is up to date, resolution step is skipped
11:17:51.110 | Progress: resolved 1, reused 0, downloaded 0, added 0
11:17:51.342 | Packages: +1021
11:17:51.342 | ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
11:17:51.873 | Packages are hard linked from the content-addressable store to the virtual store.
11:17:51.874 | Content-addressable store is at: /opt/buildhome/.local/share/pnpm/store/v3
11:17:51.874 | Virtual store is at:             node_modules/.pnpm
11:17:52.115 | Progress: resolved 1021, reused 0, downloaded 17, added 15
11:17:53.117 | Progress: resolved 1021, reused 0, downloaded 110, added 108
11:17:54.123 | Progress: resolved 1021, reused 0, downloaded 188, added 182
11:17:55.123 | Progress: resolved 1021, reused 0, downloaded 265, added 261
11:17:56.125 | Progress: resolved 1021, reused 0, downloaded 317, added 315
11:17:57.125 | Progress: resolved 1021, reused 0, downloaded 363, added 362
11:17:58.129 | Progress: resolved 1021, reused 0, downloaded 412, added 410
11:17:59.132 | Progress: resolved 1021, reused 0, downloaded 488, added 484
11:18:00.135 | Progress: resolved 1021, reused 0, downloaded 588, added 585
11:18:01.136 | Progress: resolved 1021, reused 0, downloaded 688, added 683
11:18:02.139 | Progress: resolved 1021, reused 0, downloaded 789, added 785
11:18:03.139 | Progress: resolved 1021, reused 0, downloaded 902, added 901
11:18:04.140 | Progress: resolved 1021, reused 0, downloaded 940, added 939
11:18:05.141 | Progress: resolved 1021, reused 0, downloaded 981, added 981
11:18:06.141 | Progress: resolved 1021, reused 0, downloaded 1012, added 1012
11:18:07.141 | Progress: resolved 1021, reused 0, downloaded 1016, added 1016
11:18:08.140 | Progress: resolved 1021, reused 0, downloaded 1017, added 1016
11:18:09.141 | Progress: resolved 1021, reused 0, downloaded 1018, added 1018
11:18:10.141 | Progress: resolved 1021, reused 0, downloaded 1019, added 1019
11:18:11.225 | Progress: resolved 1021, reused 0, downloaded 1020, added 1020
11:18:12.256 | Progress: resolved 1021, reused 0, downloaded 1021, added 1021, done
11:18:12.457 | .../deasync@0.1.28/node_modules/deasync install$ node ./build.js
11:18:12.512 | .../node_modules/better-sqlite3 install$ prebuild-install \|\| node-gyp rebuild --release
11:18:12.562 | .../deasync@0.1.28/node_modules/deasync install: `linux-x64-node-18` exists; testing
11:18:12.730 | .../deasync@0.1.28/node_modules/deasync install: Binary is fine; exiting
11:18:12.740 | .../deasync@0.1.28/node_modules/deasync install: Done
11:18:13.179 | .../node_modules/better-sqlite3 install: Done
11:18:13.257 | .../esbuild@0.14.47/node_modules/esbuild postinstall$ node install.js
11:18:13.259 | .../esbuild@0.16.3/node_modules/esbuild postinstall$ node install.js
11:18:13.259 | .../esbuild@0.17.6/node_modules/esbuild postinstall$ node install.js
11:18:13.259 | .../esbuild@0.15.18/node_modules/esbuild postinstall$ node install.js
11:18:13.260 | .../node_modules/workerd postinstall$ node install.js
11:18:13.386 | .../esbuild@0.14.47/node_modules/esbuild postinstall: Done
11:18:13.415 | .../esbuild@0.16.3/node_modules/esbuild postinstall: Done
11:18:13.430 | .../esbuild@0.17.6/node_modules/esbuild postinstall: Done
11:18:13.458 | .../esbuild@0.15.18/node_modules/esbuild postinstall: Done
11:18:13.593 | .../node_modules/workerd postinstall: /opt/buildhome/repo/node_modules/.pnpm/workerd@1.20230518.0/node_modules/workerd/bin/workerd: error while loading shared libraries: libc++.so.1: cannot open shared object file: No such file or directory
11:18:13.594 | .../node_modules/workerd postinstall: [workerd] Failed to validate workerd binary
11:18:13.594 | .../node_modules/workerd postinstall: Local development will not work. This usually means you're on an unsupported
11:18:13.594 | .../node_modules/workerd postinstall: operating system, or missing some shared libraries. On Debian-based systems,
11:18:13.595 | .../node_modules/workerd postinstall: make sure you've installed the `libc++1` package.
11:18:13.595 | .../node_modules/workerd postinstall: Done
11:18:14.213 | .../node_modules/vercel preinstall$ node ./scripts/preinstall.js
11:18:14.306 | .../node_modules/vercel preinstall: Done
11:18:14.655 |  
11:18:14.655 | dependencies:
11:18:14.655 | + @phosphor-icons/react 2.0.9
11:18:14.656 | + @types/node 20.3.0
11:18:14.656 | + @types/react 18.2.11
11:18:14.656 | + @types/react-dom 18.2.4
11:18:14.657 | + autoprefixer 10.4.14
11:18:14.657 | + eslint 8.42.0
11:18:14.657 | + eslint-config-next 13.4.5
11:18:14.657 | + framer-motion 10.12.16
11:18:14.657 | + next 13.4.5
11:18:14.657 | + postcss 8.4.24
11:18:14.658 | + react 18.2.0
11:18:14.658 | + react-dom 18.2.0
11:18:14.658 | + tailwindcss 3.3.2
11:18:14.659 | + typescript 5.1.3
11:18:14.659 |  
11:18:14.659 | devDependencies:
11:18:14.659 | + @cloudflare/next-on-pages 0.0.0-e48c4a0
11:18:14.659 |  
11:18:14.659 | Done in 24.2s
11:18:14.709 | Executing user command: npx @cloudflare/next-on-pages --experimental-minify
11:18:20.566 | ⚡️ @cloudflare/next-on-pages CLI v.0.0.0-e48c4a0 (beta/canary release)
11:18:20.567 |  
11:18:20.567 | ⚡️ Warning: the --experimental-minify\|-e flag is deprecated and doesn't produce any effect, the
11:18:20.567 | ⚡️ (previously named) experimental minification is now enabled by default.
11:18:20.567 | ⚡️
11:18:20.567 | ⚡️ Note: if you're using the --experimental-minify\|-e flag in your build command please remove it
11:18:20.568 | ⚡️       as it will be removed in a future version of the package (causing your command to fail).
11:18:20.568 |  
11:18:20.569 | ⚡️ Detected Package Manager: pnpm
11:18:20.569 | ⚡️ Preparing project...
11:18:20.572 | ⚡️ Project is ready
11:18:20.572 | ⚡️ Building project...
11:18:21.869 | ▲  undefined
11:18:21.870 | ▲  ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command failed with ENOENT: vercel build
11:18:21.870 | ▲  spawn vercel ENOENT
11:18:21.870 | ▲
11:18:21.870 | ▲  Command "vercel" not found.
11:18:21.904 |  
11:18:21.905 | ⚡️ The Vercel build (`pnpx vercel build`) command failed. For more details see the Vercel logs above.
11:18:21.905 | ⚡️ If you need help solving the issue, refer to the Vercel or Next.js documentation or their repositories.
11:18:21.906 |  
11:18:21.929 | Failed: Error while executing user command. Exited with error code: 1
11:18:21.938 | Failed: build command exited with code: 1
11:18:22.932 | Failed: error occurred while running build command

--- a/.changeset/breezy-ducks-knock.md
+++ b/.changeset/breezy-ducks-knock.md
@@ -1,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+fix --info showing [object Promise] for relevant packages instead of their versions

--- a/.changeset/pink-actors-learn.md
+++ b/.changeset/pink-actors-learn.md
@@ -1,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+fix vercel command not found issue

--- a/internal-packages/eslint-config-next-on-pages/index.js
+++ b/internal-packages/eslint-config-next-on-pages/index.js
@@ -1,26 +1,33 @@
 module.exports = {
-	"env": {
-		"es2021": true,
-		"node": true
+	env: {
+		es2021: true,
+		node: true
 	},
-	"extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
-	"overrides": [],
-	"parser": "@typescript-eslint/parser",
-	"parserOptions": {
-		"sourceType": "module",
-		"project": true
+	extends: ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+	overrides: [],
+	parser: "@typescript-eslint/parser",
+	parserOptions: {
+		sourceType: "module",
+		project: true
 	},
-	"plugins": ["@typescript-eslint"],
-	"rules": {
+	plugins: ["@typescript-eslint"],
+	rules: {
 		"no-case-declarations": "error",
 		"no-console": "error",
 		"prefer-const": "error",
 		"no-mixed-spaces-and-tabs": "off", // off because it conflicts with prettier
 		"eqeqeq": "error",
-
 		"@typescript-eslint/no-explicit-any": "error",
 		"@typescript-eslint/no-unused-vars": "error",
-		"@typescript-eslint/consistent-type-imports": "error"
+		"@typescript-eslint/consistent-type-imports": "error",
+
+		// Promises & async/await
+		"no-async-promise-executor": "error",
+		"no-promise-executor-return": "error",
+		"no-return-await": "error",
+		"@typescript-eslint/await-thenable": "error",
+		"@typescript-eslint/no-floating-promises": "error",
+		"@typescript-eslint/promise-function-async": "error",
 	},
-	"ignorePatterns": ["vitest.config.ts"]
+	ignorePatterns: ["vitest.config.ts"]
 };

--- a/packages/next-on-pages/src/buildApplication/buildVercelOutput.ts
+++ b/packages/next-on-pages/src/buildApplication/buildVercelOutput.ts
@@ -124,7 +124,7 @@ async function getVercelBuildChildProcess(
 async function isVercelPackageInstalled(
 	pkgMng: PackageManager
 ): Promise<boolean> {
-	const packageVersion = getPackageVersion('vercel', pkgMng);
+	const packageVersion = await getPackageVersion('vercel', pkgMng);
 	return !!packageVersion;
 }
 

--- a/packages/next-on-pages/src/cli.ts
+++ b/packages/next-on-pages/src/cli.ts
@@ -225,8 +225,8 @@ export async function printEnvInfo(): Promise<void> {
 		Package Manager Used: ${await getCurrentPackageManager()}
 		Relevant Packages:
 			@cloudflare/next-on-pages: ${nextOnPagesVersion}
-			vercel: ${getPackageVersion('vercel', packageManager) ?? 'N/A'}
-			next: ${getPackageVersion('next', packageManager) ?? 'N/A'}
+			vercel: ${await getPackageVersion('vercel', packageManager) ?? 'N/A'}
+			next: ${await getPackageVersion('next', packageManager) ?? 'N/A'}
 	`);
 
 	// eslint-disable-next-line no-console

--- a/packages/next-on-pages/src/cli.ts
+++ b/packages/next-on-pages/src/cli.ts
@@ -225,8 +225,8 @@ export async function printEnvInfo(): Promise<void> {
 		Package Manager Used: ${await getCurrentPackageManager()}
 		Relevant Packages:
 			@cloudflare/next-on-pages: ${nextOnPagesVersion}
-			vercel: ${await getPackageVersion('vercel', packageManager) ?? 'N/A'}
-			next: ${await getPackageVersion('next', packageManager) ?? 'N/A'}
+			vercel: ${(await getPackageVersion('vercel', packageManager)) ?? 'N/A'}
+			next: ${(await getPackageVersion('next', packageManager)) ?? 'N/A'}
 	`);
 
 	// eslint-disable-next-line no-console

--- a/packages/next-on-pages/src/index.ts
+++ b/packages/next-on-pages/src/index.ts
@@ -1,16 +1,16 @@
 import { watch } from 'chokidar';
 import pLimit from 'p-limit';
 import type { CliOptions } from './cli';
-import { cliWarn } from './cli';
+import { cliWarn , cliError } from './cli';
 import { cliLog, parseCliArgs, printCliHelpMessage, printEnvInfo } from './cli';
 import { buildApplication } from './buildApplication';
 import { nextOnPagesVersion } from './utils';
 
 const limit = pLimit(1);
 
-runNextOnPages();
+void runNextOnPages();
 
-function runNextOnPages(): void {
+async function runNextOnPages(): Promise<void> {
 	const args = parseCliArgs();
 
 	if (args.version) {
@@ -19,7 +19,7 @@ function runNextOnPages(): void {
 		return;
 	}
 	if (args.info) {
-		printEnvInfo();
+		await printEnvInfo();
 		return;
 	}
 
@@ -62,7 +62,7 @@ function runBuild(options: CliOptions) {
 				`);
 			}
 		}
-	});
+	}).catch(error => cliError(`Watch mode unexpected error: ${error}`));
 }
 
 function setWatchMode(fn: () => void): void {

--- a/packages/next-on-pages/src/index.ts
+++ b/packages/next-on-pages/src/index.ts
@@ -1,7 +1,7 @@
 import { watch } from 'chokidar';
 import pLimit from 'p-limit';
 import type { CliOptions } from './cli';
-import { cliWarn , cliError } from './cli';
+import { cliWarn, cliError } from './cli';
 import { cliLog, parseCliArgs, printCliHelpMessage, printEnvInfo } from './cli';
 import { buildApplication } from './buildApplication';
 import { nextOnPagesVersion } from './utils';

--- a/packages/next-on-pages/src/utils/fs.ts
+++ b/packages/next-on-pages/src/utils/fs.ts
@@ -76,7 +76,7 @@ async function validatePathType(path: string, type: 'file' | 'directory') {
  * @param path Path to check.
  * @returns Whether a file exists at the given path.
  */
-export function validateFile(path: string) {
+export async function validateFile(path: string) {
 	return validatePathType(path, 'file');
 }
 
@@ -86,7 +86,7 @@ export function validateFile(path: string) {
  * @param path Path to check.
  * @returns Whether a directory exists at the given path.
  */
-export function validateDir(path: string) {
+export async function validateDir(path: string) {
 	return validatePathType(path, 'directory');
 }
 
@@ -105,7 +105,7 @@ export async function readPathsRecursively(dir: string): Promise<string[]> {
 				const path = resolve(dir, file);
 
 				return (await validateDir(path))
-					? await readPathsRecursively(path)
+					? readPathsRecursively(path)
 					: [path];
 			})
 		);

--- a/packages/next-on-pages/src/utils/fs.ts
+++ b/packages/next-on-pages/src/utils/fs.ts
@@ -104,9 +104,7 @@ export async function readPathsRecursively(dir: string): Promise<string[]> {
 			files.map(async file => {
 				const path = resolve(dir, file);
 
-				return (await validateDir(path))
-					? readPathsRecursively(path)
-					: [path];
+				return (await validateDir(path)) ? readPathsRecursively(path) : [path];
 			})
 		);
 

